mcp_server/database.py

```
@@ -87,14 +87,48 @@ def _decode_cursor(cursor: str) -> tuple:
+            _embedding_model = model
+            _embedding_model_load_error = None
+    except Exception as exc:
+        with _embedding_model_lock:
+            _embedding_model_load_error = exc
+    finally:
+        with _embedding_model_lock:
+            _embedding_model_loading = False
+
+
+def initialize_embedding_model_async() -> None:
+    global _embedding_model_loading
```

Ruff (0.15.15)

[warning] Using the global statement to update `_embedding_model_loading` is discouraged

(PLW0603)